app/services/reservation_creator.rb

```diff
@@ -41,7 +41,7 @@ def save(session_user)
           purchase_original_cross_core_order
 
           @success = :merged_order
-        elsif @order.order_details.one?
+        elsif @order.order_details.reload.where(parent_order_detail_id: nil).one?
```

Comment:
As I understand the order_details relation is already reloaded (on rails 8 and not on 7.2) and that's why the test was failing, so I guess that part is not needed